python/tank/bootstrap/resolver.py

```diff
@@ -295,8 +296,60 @@ def _get_pipeline_configurations_for_project(self, pipeline_config_name, current
         )
 
         for pc in pipeline_configs:
+            # We'll need to provide a descriptor object for the config if
+            # possible. Note that it's possible that we'll be returning a
+            # None for the config descriptor. It's up to other filtering
+            # operations to remove those if desired.
+            #
+            # As in resolve_shotgun_configuration, the order of precedence
+            # is as follows:
+            #
+            # 1. windows/linux/mac path
+            # 2. descriptor
+            # 3. sg_descriptor
             path = ShotgunPath.from_shotgun_dict(pc)
 
+            try:
+                current_os_path = path.current_os
```

Comment:
I hadn't looked at the `current_os` implementation, so I had assumed invoking it on a platform without that path set would have thrown, but looking at the code now it seems like it throws only if you are calling it on a platform that we don't know of.

I think we can assume that this will never be the case and if it was an issue to be honest I'd rather have everything blow up and know right away than get a silent error.